cuqi/model/_model.py

```diff
@@ -160,334 +241,808 @@ def range_dim(self):
         """
         return self.range_geometry.par_dim
 
-    def _2fun(self, x, geometry, is_par):
-        """ Converts `x` to function values (if needed) using the appropriate 
-        geometry. For example, `x` can be the model input which need to be
-        converted to function value before being passed to 
-        :class:`~cuqi.model.Model` operators (e.g. _forward_func, _adjoint_func,
-        _gradient_func).
+    def _check_domain_geometry_consistent_with_forward(self):
+        """Private function that checks if the domain geometry of the model is
+        consistent with the forward operator."""
+
+        non_default_args = cuqi.utilities.get_non_default_args(self._forward_func)
```

Comment:
We have a property for this. Why repeat it here?

Reply:
updated as suggested through a new property `number_of_inputs`